src/clj/harja/palvelin/palvelut/hallinta/urakoiden_lyhytnimet.clj

```diff
@@ -0,0 +1,51 @@
+(ns harja.palvelin.palvelut.hallinta.urakoiden-lyhytnimet
+  (:require [com.stuartsierra.component :as component]
+            [taoensso.timbre :as log]
+            [harja.palvelin.komponentit.http-palvelin :refer [julkaise-palvelu poista-palvelut]]
+            [harja.domain.oikeudet :as oikeudet]
+            [harja.kyselyt.urakat :as q]))
+
+(defn- parsi-urakkatyyppi [params]
+  (let [urakkatyyppi (:urakkatyyppi params)]
+    (if (= urakkatyyppi :kaikki) (assoc {} :urakkatyyppi nil) (assoc {} :urakkatyyppi (name urakkatyyppi)))))
+
+(defn- muodosta-hakuehdot [params]
+  (merge (parsi-urakkatyyppi params) {:vain_puuttuvat (:vain-puuttuvat params) :urakantila (:urakan-tila params)}))
+(defn hae-urakoiden-nimet [db kayttaja params]
+  (let [hakuehdot (muodosta-hakuehdot params)]
+    (oikeudet/vaadi-lukuoikeus oikeudet/hallinta-indeksit kayttaja)
+    (log/debug "Haetaan urakoiden nimiä parametreilla: " params)
+    (log/debug "hakuehdot: " hakuehdot)
+    (into []
+      (q/hae-urakoiden-nimet db hakuehdot))))
+
+(defn- tallenna-urakka-nimi [db urakka]
+  (let [urakkaid (:id urakka)
+        lyhyt-nimi (:lyhyt_nimi urakka)]
+    (assert urakkaid "Urakka id puuttuu, ei voi tallentaa lyhytnimeä!")
```

Comment:
Assertti tarkistuksista on varmaan useampaa mielipidettä, mutta minusta tuo on vähän häijytapa lopettaa funktion ajo. Itse siis vältän assertin käyttämistä ja validoin sisään tulevan datan ja heitän niistä exceptionia. Assert taitaa heittää errorin.
Lainaus jostakin:
"assertions are designed to be turned off in production…they are designed to catch logically impossible situations that can only occur if the program is wrong. they should not be used to detect invalid input (like a bad web request, bad user input, etc)."

Reply:
tuon assert tarkistuksen olen tainnut apinoida jostain olemassaolevasta. Mutta heitetään tuosta poikkeus.